crates/runtime/src/memory.rs

```diff
@@ -151,7 +151,7 @@ impl RuntimeLinearMemory for MmapMemory {
         // Wasm linear memories are never allowed to grow beyond what is
         // indexable. If the memory has no maximum, enforce the greatest
         // limit here.
-        if new_pages > WASM_MAX_PAGES {
+        if new_pages > WASM_MAX_PAGES - 1 {
```

Comment:
Ideally we want to remove this restriction one day, so could this be listed as an extra check after this `if` with a `FIXME` pointing to https://github.com/bytecodealliance/wasmtime/issues/3022? That way when the issue is fixed we can simply delete the `if` without having to worry about spec correctness.